lib/Indexer/Model/Record/FullyQualifiedReferenceTrait.php

```diff
@@ -45,6 +53,11 @@ public function start(): ByteOffset
         return ByteOffset::fromInt($this->start);
     }
 
+    public function end(): ByteOffset
```

Comment:
I think, both `start` and `end` function should belong to `HasLocation` interface, as keeping it as a part of `FullyQualifiedReferenceTrait` is a bit misleading. Without this, methods aren't backed by any interface and an only reliable way to ensure, the methods are there is using assertions (which are heavily used around indexer).

Although, it may seem irrelevant, I found this itching, while working on WordPress hooks indexer, which (the hooks) are basically the first argument of a function call. It makes it a good candidate for a location data, but it doesn't have any *fqn*, what complicates the use of `FullyQualifiedReferenceTrait`

Reply:
Yeah that would definitely make sense.